CMakeLists.txt

```diff
@@ -467,32 +471,31 @@ if(ACTS_BUILD_PLUGIN_GEANT4)
     find_package(Geant4 ${_acts_geant4_version} REQUIRED CONFIG COMPONENTS gdml)
 endif()
 
-if(ACTS_BUILD_PLUGIN_TRACCC)
-    if(ACTS_USE_SYSTEM_ALGEBRAPLUGINS)
-        find_package(algebra-plugins ${_acts_algebraplugins_version} REQUIRED)
-    else()
-        add_subdirectory(thirdparty/algebra-plugins)
-    endif()
-
-    if(ACTS_USE_SYSTEM_DETRAY)
-        find_package(detray ${_acts_detray_version} REQUIRED CONFIG)
+# Convenience macro to simplify the check if we want to setup, and if
+# so, if we want to find it in the system or build internally
+macro(conditional_setup _var _package _path)
```

Comment:
Not sure if this is already done consistently for all the other functions/macros, but let's use an `acts_` prefix for this macro. (Unfortunately functions/macros enter a global namespace with CMake, so conventions like this are pretty necessary.)

Reply:
good point!